src/libraries/Common/src/System/Net/Security/Unix/SafeDeleteNegoContext.cs

```diff
@@ -42,14 +43,21 @@ public SafeGssContextHandle GssContext
         }
 
         public SafeDeleteNegoContext(SafeFreeNegoCredentials credential)
-            : base(credential)
+            : base(IntPtr.Zero)
         {
             Debug.Assert((null != credential), "Null credential in SafeDeleteNegoContext");
+            bool added = false;
+            credential.DangerousAddRef(ref added);
+            if (!added)
+            {
+                throw new ObjectDisposedException(nameof(SafeFreeNegoCredentials));
+            }
```

Comment:
`added` will never be false.  If the operation can't add, it throws an exception.
```suggestion
```